apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderIT.java

```diff
@@ -1310,11 +1310,15 @@ public void searchForNonSamhsaEobsWithExcludeSamhsaTrue() throws FHIRException {
             .execute();
     assertNotNull(searchResults);
     for (ClaimTypeV2 claimType : ClaimTypeV2.values()) {
-      // None of the claims are SAMHSA so we expect one record per claim type in the results.
-      assertEquals(
-          1,
-          filterToClaimType(searchResults, claimType).size(),
-          String.format("Verify claims of type '%s' are present", claimType));
+      /*
+       * SAMHSA fields are present on all claim types except for PDE and SNF so we should not
+       * get any claims back in the results except for PDE or SNF.
+       */
+      if (claimType == ClaimTypeV2.PDE || claimType == ClaimTypeV2.SNF) {
+        assertEquals(1, filterToClaimType(searchResults, claimType).size());
+      } else {
+        assertEquals(0, filterToClaimType(searchResults, claimType).size());
+      }
```

Comment:
I don't understand why this is coming up now with this change. Can you explain a bit more?

Reply:
The ICD-10-CM codes add SAMHSA fields. The two ITs searchForNonSamhsaEobsWithExcludeSamhsa(True & False) were affected because they use Sample A data. The ITs searchForSamhsaEobsWithExcludeSamhsa(True & False) were unaffected because these tests query manually modified Sample A data in the IT to include ICD9 SAMHSA codes, since before my changes we had non SAMHSA data for all claims.

Reply:
Upon further inspection, it appears this IT initially failed because the SAMHSA filter automatically assumed claims with the ICD-10-CM coding system URL were SAMHSA because the filter treats unknown system URLs that way. I updated the SAMHSA filter logic to check whether codes are SAMHSA for a given ICD-10-CM coding system. So the behavior of this IT should not change at all.